### PR TITLE
[charts/portal] Additional documentation for rabbitmq and tls

### PR DESCRIPTION
**Pull request by kk632390**
**Description of the change**

Added additional documentation for rabbitmq, minio and zookeeper.
Also added additional doc for tls.job and tls.rotate both for install and upgrade.

**Benefits**

**Drawbacks**

**Applicable issues**

  - fixes #

**Additional information**

**Checklist** 
- [] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [] Variables are documented in the README.md
- [] Title of the PR starts with chart name (e.g. `[charts/gateway]`)
- [] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

### CODE REVIEW

File: charts/portal/README.md [hunk 1]
```diff
@@ -563,12 +565,29 @@ Resulting hostnames:
...
+## Persistent Volumes
+With the deployment of Portal, PersistentVolumeClaim (PVC) are created for components as below:
+
+- RabbitMQ - It is used by Portal containers for internal messaging. Containers publish and/or consume messages to and from RabbitMQ.
+
+Below are for Analytics:
+
+- Kafka - Kafka is responsible to stream analytics data to druid cluster. Ingestion server is the one which streams data to Kafka, which is then ingested to druid processes. If analytics containers are not available, Kafka also act as a message store and retains Analytics data upto 6hrs. 
+
+- Zookeeper - Zookeeper is a very critical container in Druid cluster. If it is not available; its a single point of failure for the entire ingestion pipeline. Kafka and druid clusters both depend on zokeeper for sync and coordination within their respective clusters.
```
Comment (Gazza7205) [inline charts/portal/README.md]: both depend on zookeeper

File: charts/portal/README.md [hunk 2]
```diff
@@ -588,9 +607,10 @@ $ kubectl delete pvc data-rabbitmq-0|1|2
 ```
 $ kubectl scale statefulset rabbitmq --replicas=1|3
 ```
+Once the rabbitmq is running make a note of its credentials as specified in above Install Chart section 
 
 #### Your Kubernetes nodes failed or RabbitMQ crashed.
-If the RabbitMQ nodes are stopped or removed out of order, there is a chance that it won't be restored correctly.
+If the RabbitMQ cluster is stopped or removed out of order: for instance if the entire cluster loses power, there is a chance that it won't be restored correctly. Or If sync between rabbitmq peers doesn't happen or set of rabbitmq nodes can never be brought online use the 'force boot' option
```
Comment (Gazza7205) [inline charts/portal/README.md]: If the RabbitMQ cluster is stopped or removed out of order, there is a chance that it won't be restored correctly.

File: charts/portal/values-production.yaml [hunk 3]
```diff
@@ -72,6 +72,13 @@ tls:
 # If you're migrating from a docker swarm environment and wish to keep your old internal certificates
 # then follow the instructions here to load your certificates into secrets manually #### Link to doc.
```
Comment (Gazza7205) [inline charts/portal/values-production.yaml]: If you're migrating from a docker swarm environment and wish to keep your old internal certificates
then follow the instructions here to load your certificates manually - https://github.com/CAAPIM/apim-charts/blob/stable/utils/portal-migration/README.md

File: charts/portal/values-production.yaml [hunk 4]
```diff
@@ -72,6 +72,13 @@ tls:
...
 # The job will still run, but will not update anything.
+# During helm upgrade, ensure to set this enabled value to 'false' which prevents from recreating the certificates depending on rotate.
+# If the certificates are expired, during helm upgrade set this value to 'true' to create new certificates.
```
Comment (Gazza7205) [inline charts/portal/values-production.yaml:76]: The value should only be false if you intend not to run the job at all - tls.job.rotate determines upgrades.

File: charts/portal/values-production.yaml [hunk 5]
```diff
@@ -465,6 +472,7 @@ druid:
 
   minio:
     # consider changing to mode... standalone/distributed.
+    # Once Portal is installed, it is not recommented to scale up/down the minio replicas
     replicaCount: 4
```
Comment (Gazza7205) [inline charts/portal/values-production.yaml:476]: Once Portal is installed, minio can not be scaled up or down.

File: charts/portal/README.md (hunk 1 above)
Comment (kk632390) [reply charts/portal/README.md]: made changes

File: charts/portal/values-production.yaml (hunk 3 above)
Comment (kk632390) [reply charts/portal/values-production.yaml]: updated

File: charts/portal/values-production.yaml (hunk 4 above)
Comment (kk632390) [reply charts/portal/values-production.yaml:76]: Included

File: charts/portal/values-production.yaml (hunk 5 above)
Comment (kk632390) [reply charts/portal/values-production.yaml:476]: Updated

File: charts/portal/README.md (hunk 2 above)
Comment (kk632390) [reply charts/portal/README.md]: updated

File: charts/portal/README.md [hunk 6]
```diff
@@ -563,12 +565,29 @@ Resulting hostnames:
...
 | API analytics | `dev-portal-analytics.example.com` | `analytics.example.com` |
 
+## Persistent Volumes
+With the deployment of Portal, PersistentVolumeClaim (PVC) are created for components as below:
+
+- RabbitMQ - It is used by Portal containers for internal messaging. Containers publish and/or consume messages to and from RabbitMQ.
+
+Below are for Analytics:
+
+- Kafka - Kafka is responsible to stream analytics data to druid cluster. Ingestion server is the one which streams data to Kafka, which is then ingested to druid processes. If analytics containers are not available, Kafka also act as a message store and retains Analytics data upto 6hrs.
```
Comment (APIM-KiranVaddadi) [inline charts/portal/README.md]: If druid containers are not available...

Comment (kk632390) [reply charts/portal/README.md]: Updated to Druid Containers.

**Review by APIM-KiranVaddadi — APPROVED** [review]
LGTM

**Pull request merged**